src/chainstate/coordinator/mod.rs

```diff
@@ -683,6 +996,42 @@ impl<
         )
     }
 
+    // If it is time to process the exit-at-rc contract for a specific reward cycle, this function
+    // returns that reward cycle. Otherwise, the function returns None.
+    fn process_exit_reward_cycle(
+        &mut self,
+        stacks_block_height_in_cycle: u64,
+        parent_exit_info: &BlockExitRewardCycleInfo,
+    ) -> Result<Option<BlockExitRewardCycleInfo>, Error> {
+        info!(
+            "RCPR: curr height: {}, parent height: {}",
+            stacks_block_height_in_cycle, parent_exit_info.stacks_block_height_in_cycle
+        );
+        if stacks_block_height_in_cycle == 2 {
+            // We want to process the exit-at-rc contract reward cycle for the reward cycle the
+            // current block's parent's parent belongs to since the height (in number of Stacks
+            // blocks) of the current block in its reward cycle is 2.
```

Comment:
It's not clear to me why this is?  Why not process the votes in the first block of a reward cycle, or even the last block?

Reply:
Aaron felt it was a better idea to stall exit contract processing until after the first block of a reward cycle (just to avoid doing more computation in the first block of a reward cycle). 

I don't do it in the last block of the reward cycle since that would invalidate the last veto.

Reply:
> Aaron felt it was a better idea to stall exit contract processing until after the first block of a reward cycle (just to avoid doing more computation in the first block of a reward cycle).

I'm not sure the trade-off is worth it here, though (cc @kantai).  If you process the exit-at-rc logic at Stacks block height mod 2, then you have to introduce and test corner cases where you have a reward cycle with only one block.  If we can instead make the PoX reward set calculation faster (something that's useful on its own anyway, and something that we could reasonably do), then the concern about doing it in the first reward cycle block goes away, and no corner cases are needed.  This is all consensus-critical code, so if we can avoid less code even if it means a modest performance hit (or a need to spend some cycles optimizing code), then I think it's worth it.  But would love to hear a strong counter-argument.

Reply:
Let's discuss this in the architecture meeting

Reply:
I can't come this week :/  Please let me know what y'all talk about